pkg/greenlake/client.go

```diff
@@ -0,0 +1,142 @@
+// (C) Copyright 2019 Hewlett Packard Enterprise Development LP
+
+package greenlake
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Client - wrapper class for greenlake api's
+type Client struct {
+	GrantType    string
+	ClientID     string
+	ClientSecret string
+	TenantID     string
+	Host         string
+	APIKey       string
+}
+
+// Token structure
+type Token struct {
+	AccessToken     string `json:"access_token"`
+	Scope           string `json:"scope"`
+	TokenType       string `json:"token_type"`
+	Expiry          string `json:"expiry"`
+	ExpiresIn       int    `json:"expires_in"`
+	AccessTokenOnly bool   `json:"accessTokenOnly"`
+}
+
+// User structure
+type User struct {
+	Active      bool   `json:"active"`
+	DisplayName string `json:"displayName"`
+	UserName    string `json:"userName"`
+	Name        Name   `json:"name"`
+}
+
+// Name structure
+type Name struct {
+	FamilyName string `json:"familyName"`
+	GivenName  string `json:"givenName"`
+}
+
+// NewGreenLakeClient create
+func NewGreenLakeClient(grantType, clientID, secretKey, tenantID, host string) *Client {
+	return &Client{
+		GrantType:    grantType,
+		ClientID:     clientID,
+		ClientSecret: secretKey,
+		TenantID:     tenantID,
+		Host:         host,
+		APIKey:       "",
+	}
+}
+
+// NewGLClientFromAPIKey creates a new GreenLake Client from existing API sessions key
+func NewGLClientFromAPIKey(host, tenantID, apikey string) *Client {
+	return &Client{
+		GrantType:    "client_credentials",
+		ClientID:     "",
+		ClientSecret: "LOCAL",
+		APIKey:       apikey,
+		TenantID:     tenantID,
+		Host:         host,
+	}
+}
+
+// GetToken api
+func (c *Client) GetToken() (Token, error) {
+	var result Token
+
+	url := fmt.Sprintf(c.Host + "/identity/v1/token")
```

Comment:
Wrong usage of Sprintf
you can either concat the string directly - then there is no need for fmt.Sprintf
or you need a template string in sprintf, which would be something like:
url := fmt.Sprintf("%s%s, c.Host, "/identity/vi/token")